src/roi/Roi.ts

```diff
@@ -328,6 +329,29 @@ export class Roi {
     };
   }
 
+  get centroid() {
+    return this.#getComputed('centroid', () => {
+      const roiMap = this.getMap();
+      const data = roiMap.data;
+      let xCoords = 0;
+      let yCoords = 0;
+      for (let column = 0; column < this.width; column++) {
+        for (let row = 0; row < this.height; row++) {
+          let target = this.computeIndex(row, column);
+          if (data[target] === this.id) {
+            xCoords += column + this.origin.column;
```

Comment:
The code is better but adding the origin can be done only once when returning the value